ctapipe/image/ellipsoid.py

```diff
@@ -0,0 +1,407 @@
+"""
+
+from enum import Enum
+
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import Angle
+from iminuit import Minuit
+
+from ctapipe.image.cleaning import dilate
+from ctapipe.image.hillas import hillas_parameters
+from ctapipe.image.pixel_likelihood import (
+    mean_poisson_likelihood_gaussian,
+    neg_log_likelihood_approx,
+)
+from ctapipe.image.toymodel import SkewedGaussian
+
+from ..containers import CameraImageFitParametersContainer, ImageFitParametersContainer
+
+PED_TABLE = {
+    "LSTCam": 2.8,
+    "NectarCam": 2.3,
+    "FlashCam": 2.3,
+    "SSTCam": 0.5,
+    "CHEC": 0.5,
+    "DUMMY": 0,
+    "testcam": 0,
+}
+SPE_WIDTH = 0.5
+
+__all__ = [
+    "image_fit_parameters",
+    "ImageFitParameterizationError",
+    "PDFType",
+]
+
+
+class PDFType(Enum):
+    gaussian = "gaussian"
+    skewed = "skewed"
+
+
+def create_initial_guess(geometry, hillas, size):
+    """
+
+    This function computes the seeds of the fit with the Hillas parameters
+
+    Parameters
+    ----------
+    geometry: ctapipe.instrument.CameraGeometry
+        Camera geometry, the cleaning mask should be applied to improve performance
+    hillas: HillasParametersContainer
+        Hillas parameters
+    size: float/int
+        Total charge after cleaning and dilation
+
+    Returns
+    -------
+    initial_guess: dict
+    Seed parameters of the fit.
+    """
+    unit = geometry.pix_x.unit
+    initial_guess = {}
+
+    if unit.is_equivalent(u.m):
+        initial_guess["cog_x"] = hillas.x.to_value(unit)
+        initial_guess["cog_y"] = hillas.y.to_value(unit)
+    else:
+        initial_guess["cog_x"] = hillas.fov_lon.to_value(unit)
+        initial_guess["cog_y"] = hillas.fov_lat.to_value(unit)
+
+    initial_guess["length"] = hillas.length.to_value(unit)
+    initial_guess["width"] = hillas.width.to_value(unit)
+    initial_guess["psi"] = hillas.psi.value
+
+    initial_guess["skewness"] = hillas.skewness
+
+    if (hillas.width.to_value(unit) == 0) or (hillas.length.to_value(unit) == 0):
+        raise ImageFitParameterizationError("Hillas width and/or length is zero")
+
+    initial_guess["amplitude"] = size
+    return initial_guess
+
+
+def boundaries(geometry, image, dilated_mask, hillas, pdf):
+    """
+
+    Computes the boundaries of the fit.
+
+    Parameters
+    ----------
+    geometry : ctapipe.instrument.CameraGeometry
+        Camera geometry
+    image : ndarray
+        Charge in each pixel, no cleaning mask should be applied
+    dilated_mask : ndarray
+        mask (array of booleans) after image cleaning and dilation
+    hillas : HillasParametersContainer
+       Hillas parameters
+    pdf: PDFType instance
+        e.g. PDFType("gaussian")
+
+    Returns
+    -------
+    list of boundaries
+    """
+    x = geometry.pix_x.value
+    y = geometry.pix_y.value
+    unit = geometry.pix_x.unit
+    camera_radius = geometry.radius.to_value(unit)
+
+    # Using dilated image
+    row_image = image.copy()
```

Comment:
Why do you set the pixels to 0 in a copy of the image instead of just using the surviving pixels like we e.g. do with the hillas parameters and the muon fit?